decidim-core/lib/decidim/icon_registry.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Decidim
+  class IconRegistry
+    def initialize
+      @icons = ActiveSupport::HashWithIndifferentAccess.new
+    end
+
+    # Registers a new icon.
+    #
+    # @param name [String] The name of the icon. It will be used to find the icon later.
+    # @param icon [String] The id of the icon. It will be used to load the icon from remixicon library.
+    # @param resource [String] The resource name. The resource name. It will be used to link the icon to a specific resource.
+    # @param category [String] The category name. It will be used to group the icons by category.
+    # @param description [String] The description of the icon. It will be used to show the purpose of the icon in DDG.
+    # @param engine [String] The engine name.It is used internally to identify the module where the icon is being used.
+    def register(name:, icon:, resource:, description:, category:, engine:) # rubocop:disable Metrics/ParameterLists
```

Comment:
I see I'm not the only one that thinks that these are too many parameters :trollface:

Reply:
I have stated what is the purpose of each one of the params. If you still think i have an actionable here, i may be able to "hide" 2 fields ( resource, category and engine) and have a contract like this: 

```
    def register(name:, icon:, resource:, description:, options = {})
```
but that would complicate the DDG part.

Reply:
I have removed resource parameter as it did not made any sense. [1288dc9](https://github.com/decidim/decidim/pull/11982/commits/1288dc90e78c9d3609ed048e04811e9ce1dbcb18)